go/stats/export.go

```diff
@@ -30,24 +30,41 @@ package stats
 import (
 	"bytes"
 	"expvar"
-	"flag"
 	"fmt"
 	"strconv"
 	"strings"
 	"sync"
 	"time"
 
+	"github.com/spf13/pflag"
+
 	"vitess.io/vitess/go/vt/log"
 )
 
-var emitStats = flag.Bool("emit_stats", false, "If set, emit stats to push-based monitoring and stats backends")
-var statsEmitPeriod = flag.Duration("stats_emit_period", time.Duration(60*time.Second), "Interval between emitting stats to all registered backends")
-var statsBackend = flag.String("stats_backend", "", "The name of the registered push-based monitoring/stats backend to use")
-var combineDimensions = flag.String("stats_combine_dimensions", "", `List of dimensions to be combined into a single "all" value in exported stats vars`)
-var dropVariables = flag.String("stats_drop_variables", "", `Variables to be dropped from the list of exported variables.`)
+var (
+	emitStats         bool
+	statsEmitPeriod   = 60 * time.Second
+	statsBackend      string
+	combineDimensions string
+	dropVariables     string
+)
 
 // CommonTags is a comma-separated list of common tags for stats backends
-var CommonTags = flag.String("stats_common_tags", "", `Comma-separated list of common tags for the stats backend. It provides both label and values. Example: label1:value1,label2:value2`)
+var CommonTags string
+
+func init() {
+	registerFlags()
```

Comment:
Should'nt we do `servenv.OnParseFor(registerFlags)` even if we want these flags to be on all the binaries that import this package? Same for other places too where we call registerFlags in init directly

Reply:
The `stats` package is used by a lot of binaries, hence I just called `RegisterFlags()` so that it is available to all binaries. What is the best approach, @ajm188? 
* listing all binaries to `servenv.OnParseFor()`. I was afraid of breaking something if I did this, for example I am not sure if ...
* add `stats` to `servenv.init()` like we do for `log` or `logutil`

```
$ go list -f '{{ .ImportPath }}| {{ join .Deps " " }} ' ./go/cmd/... | awk -F"|" '$2 ~ "vitess.io/vitess/go/stats" {print $1}'
vitess.io/vitess/go/cmd/mysqlctl
vitess.io/vitess/go/cmd/mysqlctld
vitess.io/vitess/go/cmd/query_analyzer
vitess.io/vitess/go/cmd/rulesctl
vitess.io/vitess/go/cmd/rulesctl/cmd
vitess.io/vitess/go/cmd/rulesctl/common
vitess.io/vitess/go/cmd/topo2topo
vitess.io/vitess/go/cmd/vtaclcheck
vitess.io/vitess/go/cmd/vtadmin
vitess.io/vitess/go/cmd/vtbackup
vitess.io/vitess/go/cmd/vtbench
vitess.io/vitess/go/cmd/vtclient
vitess.io/vitess/go/cmd/vtcombo
vitess.io/vitess/go/cmd/vtctl
vitess.io/vitess/go/cmd/vtctlclient
vitess.io/vitess/go/cmd/vtctld
vitess.io/vitess/go/cmd/vtctldclient
vitess.io/vitess/go/cmd/vtctldclient/cli
vitess.io/vitess/go/cmd/vtctldclient/command
vitess.io/vitess/go/cmd/vtctldclient/docgen
vitess.io/vitess/go/cmd/vtexplain
vitess.io/vitess/go/cmd/vtgate
vitess.io/vitess/go/cmd/vtgateclienttest
vitess.io/vitess/go/cmd/vtgr
vitess.io/vitess/go/cmd/vtorc
vitess.io/vitess/go/cmd/vttablet
vitess.io/vitess/go/cmd/vttestserver
vitess.io/vitess/go/cmd/zk
```

Reply:
My understanding is that we should use `servenv.OnParse` which doesn't take in a binary at all, just the function to register and registers it for all the binaries that import this package. 

PS - Sorry, I wanted to type `OnParse` in the original message too and not `OnParseFor`

Reply:
Ah, I realized why I had not used `servenv` in `go/stats`. `servenv` itself imports `go/stats` leading to a import cycle.